### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by bryanrichie**

### CODE REVIEW

File: src/client/components/Comments.tsx [hunk 1]
```diff
@@ -1,8 +1,13 @@
-import { VStack, Text } from '@chakra-ui/react';
-import { VscCommentDiscussion } from 'react-icons/vsc';
+import { Text, VStack } from '@chakra-ui/react';
 import React from 'react';
+import { VscCommentDiscussion } from 'react-icons/vsc';
+import { useParams } from 'react-router-dom';
+import { useGetComments } from '../hooks/useGetComments';
 
 export const Comments = () => {
+  const { postId } = useParams<{ postId: string }>();
```
Comment (dbousamra) [inline src/client/components/Comments.tsx]: I have a util hook:

```
import { useParams } from 'react-router-dom';

export function useRequiredParams<A>() {
  return (useParams() as any) as Required<A>;
}

```

File: src/server/main.ts [hunk 2]
```diff
@@ -117,41 +117,64 @@ app.post('/post/create', authMiddleware, async (req: Request, res, next) => {
   }
 });
 
-app.post('/post/delete', async (req, res, next) => {
+// app.post('/post/delete', async (req, res, next) => {
```
Comment (dbousamra) [inline src/server/main.ts:110]: Commented out code

File: src/server/main.ts [hunk 3]
```diff
@@ -117,41 +117,64 @@ app.post('/post/create', authMiddleware, async (req: Request, res, next) => {
...
-    res.status(200).json({ deletedPost, status: 'Post successfully deleted.' });
+    res.status(200).json({ post });
   } catch (error) {
     next(error);
   }
 });
 
-app.post('/post/edit', async (req, res, next) => {
+app.post('/posts/:postId/comments', authMiddleware, async (req: Request, res, next) => {
+  console.log(req.body);
```
Comment (dbousamra) [inline src/server/main.ts]: Leftover Console.log

File: src/server/middleware/error.middleware.ts [hunk 4]
```diff
@@ -10,6 +10,8 @@ export const errorMiddleware = (
   next: NextFunction
 ) => {
   if (!isCustomError(error)) {
+    console.log('ERR', error);
```
Comment (dbousamra) [inline src/server/middleware/error.middleware.ts]: Left over

File: src/client/components/CommentsForm.tsx [hunk 5]
```diff
@@ -1,28 +1,26 @@
...
-  text?: string;
+  comment: string;
 }
 
 const validationSchema = yup
   .object()
   .shape({
-    title: yup.string().max(300).required(),
-    text: yup.string().optional(),
+    comment: yup.string().max(10000).required(),
```
Comment (willemclarke) [inline src/client/components/CommentsForm.tsx:18]: I learned yesterday doing piccolo work that you don't need to call `.shape(...)` after `.object`
you can simply go 
```
const validationSchema = yup
  .object({
    comment: yup.string().max(10000).required()
  }).required()

 ```

File: src/client/components/Comments.tsx (hunk 1 above)
Comment (benjiheath) [reply src/client/components/Comments.tsx]: :hehe:

**Pull request merged**